frame/balances/src/benchmarking.rs

```diff
@@ -176,6 +176,22 @@ benchmarks_instance_pallet! {
 		assert_eq!(Balances::<T, I>::free_balance(&caller), Zero::zero());
 		assert_eq!(Balances::<T, I>::free_balance(&recipient), transfer_amount);
 	}
+
+	// Benchmark `transfer_all` with the worst possible condition:
+	// * The recipient account is created
+	// * The sender is killed
+	transfer_all {
+		let caller = whitelisted_caller();
+		let recipient: T::AccountId = account("recipient", 0, SEED);
+		let recipient_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(recipient.clone());
+
+		// Give the sender account max funds, thus a transfer will not kill account.
+		let _ = <Balances<T, I> as Currency<_>>::make_free_balance_be(&caller, T::Balance::max_value());
```

Comment:
This is really confusing to me.
What do you mean with "thus a transfer will not kill account"? Isn't that the point of `transfer_all`, to kill the sender?
Do you mean to make sure that the recipient is not killed?

Reply:
yeah this is copy paste, sorry